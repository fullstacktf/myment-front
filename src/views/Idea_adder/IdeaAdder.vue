<template></template>

<script>
export default {
  name: IdeaAdder,
};
</script>

<style>
</style>
