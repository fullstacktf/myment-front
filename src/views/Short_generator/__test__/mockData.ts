import { MockCity, MockIdea, MockCategory, MockZone } from "./mockTypes";

export interface MockData implements MockCity{
    city1: MockCity,
    city2: MockCity
}


export const mockData: MockCity = 
{
    city:[{
            lodging: {
                color: "orange",
                ideas:[{
                    name: "idea name",
                    description: "lorem ipsum dolor sit amet ...",
                    startTime:09.00,
                    endTime:12.00,
                    link: "http://..."
                }]
            },
            food: {
                color: "green",
                    ideas: [{
                            name: "idea name",
                            description: "lorem ipsum dolor sit amet ...",
                            startTime:09.00,
                            endTime:12.00,
                            link: "http://..."
                        }]
            },
            leisure: {
                color: "orange",
                ideas: [{
                    name: "idea name",
                    description: "lorem ipsum dolor sit amet ...",
                    startTime:09.00,
                    endTime:12.00,
                    link: "http://..."
                }]
            }
        },	
        {
            lodging: {
                color: "orange",
                ideas:[{
                    name: "idea name",
                    description: "lorem ipsum dolor sit amet ...",
                    startTime:09.00,
                    endTime:12.00,
                    link: "http://..."
                }]
            },
            food: {
                color: "green",
                    ideas: [{
                            name: "idea name",
                            description: "lorem ipsum dolor sit amet ...",
                            startTime:09.00,
                            endTime:12.00,
                            link: "http://..."
                        }]
            },
            leisure: {
                color: "orange",
                ideas: [{
                    name: "idea name",
                    description: "lorem ipsum dolor sit amet ...",
                    startTime:09.00,
                    endTime:12.00,
                    link: "http://..."
                }]
            }
        },	
    ],
    coordinates:{
        "type": "FeatureCollection",
        "features": [
          {
            "type": "Feature",
            "properties": {},
            "geometry": {
              "type": "Point",
              "coordinates": [
                -16.31916046142578,
                28.483478824137844
              ]
            }
          }
        ]
      }
}
