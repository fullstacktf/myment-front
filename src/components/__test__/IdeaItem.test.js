import IdeaAdder from '../IdeaItem.vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Buefy from 'buefy';
import { mockData } from '../../test/mockData';

const mock = {
  name: mockData.zone[0].food.ideas[0].name,
  description: mockData.zone[0].food.ideas[0].description,
  start: mockData.zone[0].food.ideas[0].startTime[0],
  end: mockData.zone[0].food.ideas[0].endTime[0],
};
describe('IdeaAdder', () => {
  let localVue;
  beforeEach(() => {
    const Vue = createLocalVue();
    Vue.use(Buefy);
    localVue = Vue;
  });
  test('render component', () => {
    const wrapper = shallowMount(IdeaAdder, {
      localVue,
      propsData: {
        ...mock,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('check button props', () => {
    const wrapper = shallowMount(IdeaAdder, {
      localVue,
      propsData: {
        ...mock,
      },
    });
    const button = wrapper.find('button');
    expect(button.text()).toBe('Guachinche Ivan');
  });

  test('check Contents props', () => {
    const wrapper = shallowMount(IdeaAdder, {
      localVue,
      propsData: {
        ...mock,
      },
    });
    const content = wrapper.find('.content');
    const Time = content.findAll('h3');
    const description = content.find('p');

    expect(Time.at(0).text()).toBe('9');
    expect(Time.at(1).text()).toBe('11');
    expect(description.text()).toBe(
      'Go to Guachinche Ivan in La Laguna and take a rabbit',
    );
  });
});
