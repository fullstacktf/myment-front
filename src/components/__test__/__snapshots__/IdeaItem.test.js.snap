// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaAdder render component 1`] = `
"<div class=\\"ideaWrapper\\"><b-collapse-stub animation=\\"fade\\" ariaid=\\"contentIdForA11y1\\"><button aria-controls=\\"contentIdForA11y1\\" class=\\"button is-dark is-rounded is-large is-fullwidth is-focusable\\"></button> <div class=\\"notification\\"><div class=\\"content\\"><div class=\\"starttime\\"><span></span> :
          <span></span></div> <p>no description added</p> <a href=\\"#\\" target=\\"_blank\\" class=\\"button is-info\\">location</a> <div class=\\"endtime\\"><span></span> :
          <span></span></div></div></div></b-collapse-stub></div>"
`;
